sdk/python/lib/pulumi/runtime/invoke.py

```diff
@@ -120,6 +120,18 @@ def do_invoke():
 
         # Otherwise, return the output properties.
         ret_obj = getattr(resp, "return")
+
+        # To avoid breaking older versions of the Kubernetes Python SDK, if the result of
+        # the invoke is not truthy (e.g. it's an empty struct), and the token is one of the
+        # known kubernetes invoke tokens, return None instead of an empty dict.
+        # See https://github.com/pulumi/pulumi/issues/14508.
+        if not ret_obj and tok in {
```

Comment:
Should we check against version here so new k8s providers get the correct behavior of an empty struct? I guess that version isn't released yet but maybe assume next major and we can reduce it if providers fix it sooner?